api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirTaskServiceImpl.java

```diff
@@ -40,9 +40,25 @@ public Task getTaskByUuid(String uuid) {
 	 */
 	@Override
 	public Task saveTask(Task task) {
-		org.openmrs.module.fhir2.Task savedOpenmrsTask = dao.saveTask(translator.toOpenmrsType(task));
+		return translator.toFhirResource(dao.saveTask(translator.toOpenmrsType(task)));
+	}
+	
+	/**
+	 * Save task to the DB, or update task if one exists with given UUID
+	 * 
+	 * @param uuid the uuid of the task to update
+	 * @param task the task to save
+	 * @return the saved task
+	 */
+	@Override
+	public Task updateTask(String uuid, Task task) {
```

Comment:
Don't  you guys think that maybe we can stick to the same approach being used in openmrs core and openmrs modules where same save method is used for both save and update?

Reply:
It's okay to have the same save method for saveOrUpdate in the **`DAO`** layer. But we should try to distinguish in the **`service layer`**. Although it's using the same dao method.